src/main/java/Parser.java

```diff
@@ -40,4 +41,64 @@ public static String printDate(LocalDateTime dateObj) {
                 + String.format("%02d", dateObj.getMinute());
     }
 
+    public static int parseTaskInt(String str, List<Task> taskList) throws DukeException {
+        int taskInt;
+        try {
+            taskInt = Integer.parseInt(str);
+            if (taskInt > taskList.size() || taskInt <= 0) {
+                throw new DukeException("That task number does not exist, please try again");
```

Comment:
Throwing a new `NumberFormatException` here can help reduce the number of code paths to maintain.

Reply:
I coded it this way as I felt there was a difference between the one cited above (where the input number is valid, but it doesnt correspond to a valid task number) whereas when a NumberFormatException is thrown, it is because something other than an integer is typed and parsed